Description: Stop using non-DFSG gpuocelot code

backend/src/ir/structural_analysis.* (derived from gpuocelot) have
been deleted because they are not DFSG-free:
https://lists.debian.org/debian-legal/2015/05/msg00008.html

Origin: upstream 347116ef2855e73c25dc09c0a8ee1c7f58c46099
Author: Zhigang Gong <zhigang.gong@intel.com>

diff --git a/backend/src/CMakeLists.txt b/backend/src/CMakeLists.txt
index a6736ec..45f18d8 100644
--- a/backend/src/CMakeLists.txt
+++ b/backend/src/CMakeLists.txt
@@ -66,8 +66,6 @@ set (GBE_SRC
     ir/lowering.hpp
     ir/printf.cpp
     ir/printf.hpp
-    ir/structural_analysis.cpp
-    ir/structural_analysis.hpp
     ir/immediate.hpp
     ir/immediate.cpp
     backend/context.cpp
diff --git a/backend/src/llvm/llvm_to_gen.cpp b/backend/src/llvm/llvm_to_gen.cpp
index a4ce4a2..3e11fdb 100644
--- a/backend/src/llvm/llvm_to_gen.cpp
+++ b/backend/src/llvm/llvm_to_gen.cpp
@@ -61,7 +61,6 @@
 #include "sys/cvar.hpp"
 #include "sys/platform.hpp"
 #include "ir/unit.hpp"
-#include "ir/structural_analysis.hpp"
 
 #include <clang/CodeGen/CodeGenAction.h>
 
@@ -308,16 +307,6 @@ namespace gbe
     // Print the code extra optimization passes
     OUTPUT_BITCODE(AFTER_GEN, mod);
 
-    const ir::Unit::FunctionSet& fs = unit.getFunctionSet();
-    ir::Unit::FunctionSet::const_iterator iter = fs.begin();
-    while(iter != fs.end())
-    {
-      analysis::ControlTree *ct = new analysis::ControlTree(iter->second);
-      ct->analyze();
-      delete ct;
-      iter++;
-    }
-
     delete libraryInfo;
     return true;
   }

